app/src/main/java/com/example/habitsmasher/MainActivity.java

```diff
@@ -53,6 +57,26 @@ protected void onCreate(Bundle savedInstanceState) {
             NavigationUI.setupWithNavController(_binding.navView, _navController);
             _navController.navigate(R.id.user_login);
         }
+
+        setOnDestinationChangedListenerForNavController();
+    }
+
+    private void setOnDestinationChangedListenerForNavController() {
+        _navController.addOnDestinationChangedListener(new NavController.OnDestinationChangedListener() {
+            @Override
+            public void onDestinationChanged(@NonNull NavController controller,
+                                             @NonNull NavDestination destination,
+                                             @Nullable Bundle arguments) {
+                View bottomNav = findViewById(R.id.nav_view);
+                if(destination.getId() == R.id.user_login ||
+                    destination.getId() == R.id.user_registration) {
+                    bottomNav.setVisibility(View.GONE);
+                } else {
+                    bottomNav.setVisibility(View.VISIBLE);
+                }
+
+            }
+        });
```

Comment:
This method sets the destination changed listener for the nav controller. Basically, it is responsible for checking which fragment we are currently on, and hiding it if its the **login or user sign up** pages. Otherwise, we want to be showing the bottom navigation bar.